lib/parser/create.js

```diff
@@ -315,21 +315,29 @@ export function createParser(config) {
         parser.parseValue = 'parseValue' in options ? Boolean(options.parseValue) : true;
         parser.parseCustomProperty = 'parseCustomProperty' in options ? Boolean(options.parseCustomProperty) : false;
 
-        const { context = 'default', onComment } = options;
+        const { context = 'default', onComment, onToken } = options;
 
         if (context in parser.context === false) {
             throw new Error('Unknown context `' + context + '`');
         }
 
-        if (typeof onComment === 'function') {
+        if (typeof onComment === 'function' || typeof onToken === 'function' || Array.isArray(onToken)) {
```

Comment:
Nit: `onToken` can be checked for `isArray()` before the block and turned into a function if needed. This allows to simplify things, since for the block `onToken` will be a function or undefined.

```js
if (Array.isArray(onToken)) {
    onToken = (type, start, end) => ({ type, start, end });
}
```